DigitalLearningSolutions.Web/ViewModels/Register/RegisterDelegateByCentre/WelcomeEmailViewModel.cs

```diff
@@ -22,7 +24,17 @@ public WelcomeEmailViewModel(DelegateRegistrationByCentreData data)
         public int? Month { get; set; }
         public int? Year { get; set; }
         public bool ShouldSendEmail { get; set; }
-        public DateValidator.ValidationResult? DateValidationResult { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (!ShouldSendEmail)
+            {
+                return new List<ValidationResult>();
+            }
+
+            var dateValidationResult = DateValidator.ValidateDate(Day, Month, Year, "Email delivery date", true);
+            return DateValidator.ToValidationResultList(dateValidationResult, "Day", "Month", "Year");

```

Comment:
Instead of hardcoded strings, should we use property names for "Day", "Month", and "Year"?